fantasybot/telegram/bot.py

```
@@ -0,0 +1,1034 @@
+                    optimal_ids.add(str(s_p["playerTeamId"]))
+
+                if current_ids == optimal_ids:
+                    lines.append("\n✅ <b>¡Tu XI actual ya es el óptimo!</b>")
+                    lines.append("<i>No necesitas realizar ningún cambio para la próxima jornada.</i>")
+                else:
+                    can_apply = True
+                    lines.append(f"\n🌟 <b>Alineación Óptima Recomendada ({opt_tactical})</b>\n")
+                    gk_name = best["goalkeeper"].get("nombre") or best["goalkeeper"].get("name")
+                    lines.append(f"🧤 <b>PORTERO:</b> <b>{gk_name}</b>")
+                    defs = [p.get("nombre") or p.get("name") for p in best.get("defender", [])]
+                    lines.append(f"🛡 <b>DEFENSAS:</b> " + " • ".join(f"<b>{n}</b>" for n in defs))
```

Ruff (0.16.1)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)